src/arkode/arkode_arkstep_io.c

```diff
@@ -781,6 +790,51 @@ int arkStep_SetNonlinear(ARKodeMem ark_mem)
   return (ARK_SUCCESS);
 }
 
+/*---------------------------------------------------------------
+  arkStep_SetAutonomous:
+
+  Indicates if the problem is autonomous (True) or non-autonomous
+  (False).
+  ---------------------------------------------------------------*/
+int arkStep_SetAutonomous(ARKodeMem ark_mem, sunbooleantype autonomous)
+{
+  ARKodeARKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeARKStepMem structure */
+  retval = arkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  step_mem->autonomous = autonomous;
+
+  if (autonomous && step_mem->linear) { step_mem->linear_timedep = SUNFALSE; }
+
+  /* Reattach the nonlinear system function e.g., switching to/from an
+     autonomous problem with the trivial predictor requires swapping the
+     nonlinear system function provided to the nonlinear solver */
+  retval = arkStep_SetNlsSysFn(ark_mem);
+  if (retval != ARK_SUCCESS)
+  {
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Setting nonlinear system function failed");
+    return ARK_ILL_INPUT;
+  }
+
+  /* This will be better handled when the temp vector stack is added */
+  if (autonomous)
+  {
+    /* Allocate tempv5 if needed */
+    if (!arkAllocVec(ark_mem, ark_mem->yn, &ark_mem->tempv5))
+    {
+      arkProcessError(ark_mem, ARK_MEM_FAIL, __LINE__, __func__, __FILE__,
+                      MSG_ARK_MEM_FAIL);
+      return ARK_MEM_FAIL;
+    }
+  }
```

Comment:
Should this routine also free `ark_mem->tempv5` in the case that `autonomous` is false, e.g., if a user called this twice, first with `autonomous` set to true and then with `autonomous` set to false?

Reply:
Yes, it would be safe to free that vector here.